com.vrcfury.vrcfury/Editor/VF/Inspector/VRCFuryActionEditor.cs

```diff
@@ -108,6 +108,126 @@ public class VRCFuryActionDrawer : PropertyDrawer {
 
                 return row;
             }
+            case nameof(MaterialPropertyAction): {
+                var row = new VisualElement {
+                    style = {
+                        flexDirection = FlexDirection.Row,
+                        alignItems = Align.FlexStart
+                    }
+                };
+
+                var label = new Label("Material Property") {
+                    style = {
+                        flexGrow = 0,
+                        flexBasis = 100
+                    }
+                };
+
+                var col = new VisualElement
+                {
+                    style =
+                    {
+                        flexDirection = FlexDirection.Column,
+                        flexGrow = 1
+                    }
+                };
+                
+                row.Add(label);
+
+                var rendererRow = new VisualElement {
+                    style = {
+                        flexDirection = FlexDirection.Row,
+                        alignItems = Align.Center,
+                        marginBottom = 1
+                    }
+                };
+
+                var affectAllMeshesProp = prop.FindPropertyRelative("affectAllMeshes");
+                
+                var rendererProp = prop.FindPropertyRelative("renderer");
+                var propField = VRCFuryEditorUtils.Prop(rendererProp);
+                propField.style.flexGrow = 1;
+                propField.style.flexShrink = 1;
+                propField.SetEnabled(!affectAllMeshesProp.boolValue);
+                rendererRow.Add(propField);
+                
+                rendererRow.Add(new Label("All Meshes") {
+                    style = {
+                        marginLeft = 2,
+                        marginRight = 2,
+                        flexGrow = 1,
+                        flexBasis = 100,
+                        unityTextAlign = TextAnchor.MiddleRight
+                    }
+                });
+                
+                var propField4 = VRCFuryEditorUtils.Prop(affectAllMeshesProp);
+                propField4.style.flexGrow = 0;
+                propField4.style.flexShrink = 0;
+                propField4.style.flexBasis = 16;
+                propField4.schedule.Execute(() => {

```

Comment:
Does this automatically cancel the schedule when the visualelement is disposed or detached or something?

Reply:
Also, does the callback run in the main thread?

Reply:
We wrote basically something just like this [here](https://github.com/VRCFury/VRCFury/blob/main/com.vrcfury.vrcfury/Editor/VF/Inspector/VRCFuryEditorUtils.cs#L715), but I'd be happy to replace it if this does the same.

Reply:
The callback does run on main, and it does de-schedule when the element is removed.

In the particular case of what I'm trying to do there - in 2022 they added a [`TrackPropertyValue`](https://docs.unity3d.com/ScriptReference/UIElements.BindingExtensions.TrackPropertyValue.html) extension which sets up the loop for you automatically so its a bit cleaner too. But in 2019 - the `schedule.Execute().Every()` is the best approach I found

Reply:
We have a callback for that already called `VRCFuryEditorUtils.RefreshOnChange`. It's usually designed to re-render the entire inside of the callback, but you can also just return an empty VisualElement and then do whatever you want in the callback.

Reply:
Migrated to `RefreshOnChange`